src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,115 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+
+import { Person } from '../types/Person';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function debounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+type Props = {
+  people: Person[];
+  onSelect?: React.Dispatch<React.SetStateAction<Person | null>>;
+  delay: number;
+  selectedPerson: Person | null;
+};
+
+export const Autocomplete: React.FC<Props> = React.memo(
+  ({
+    people,
+    onSelect = () => {},
+    delay,
+    selectedPerson,
+  }) => {
+    const [query, setQuery] = useState('');
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+    const [appliedQuery, setAppliedQuery] = useState('');
+
+    const applyQuery = useCallback(() => debounce(
+      setAppliedQuery, delay,
+    ), [setAppliedQuery, delay]);
+
+    const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/4a8f8efc-269c-4498-bee2-744c54c9c19b)

once you type anything set  `isDropdownOpen` false, and after delay set it `true`

P.S. It's better to ask for help in the chat - the response will be faster